module/zfs/vdev_raidz.c

```diff
@@ -3141,9 +3217,12 @@ raidz_reflow_impl(vdev_t *vd, vdev_raidz_expand_t *vre, range_tree_t *rt,
 	    ZIO_FLAG_CANFAIL,
 	    raidz_reflow_write_done, rra);
 
+	roffset = (blkid / old_children) << ashift;
+	if (vre->vre_scratch_devices != 0)
+		roffset -= VDEV_BOOT_SIZE;
```

Comment:
Because of next expression in the zio_vdev_child_io():
	if (vd->vdev_ops->vdev_op_leaf) {
		ASSERT0(vd->vdev_children);
		offset += VDEV_LABEL_START_SIZE;
	}